core/trino-main/src/test/java/io/trino/type/TestUuidType.java

```diff
@@ -61,4 +73,55 @@ public void testDisplayName()
     {
         assertEquals(UUID.getDisplayName(), "uuid");
     }
+
+    @Test
+    public void testJavaUuidToTrinoUuid()
+    {
+        assertThat(javaUuidToTrinoUuid(java.util.UUID.fromString("00000000-0000-0000-0000-000000000001")))
+                .isEqualTo(castFromVarcharToUuid(utf8Slice("00000000-0000-0000-0000-000000000001")));
+
+        assertThat(javaUuidToTrinoUuid(java.util.UUID.fromString("f79c3e09-677c-4bbd-a479-3f349cb785e7")))
+                .isEqualTo(castFromVarcharToUuid(utf8Slice("f79c3e09-677c-4bbd-a479-3f349cb785e7")));
+    }
+
+    @Test
+    public void testOrdering()
+            throws Throwable
+    {
+        String lowerAsString = "406caec7-68b9-4778-81b2-a12ece70c8b1";
+        String higherAsString = "f79c3e09-677c-4bbd-a479-3f349cb785e7";
+        java.util.UUID lower = java.util.UUID.fromString(lowerAsString);
+        java.util.UUID higher = java.util.UUID.fromString(higherAsString);
+        // Java UUID's comparison is not consitent with RFC 4122, see https://bugs.openjdk.org/browse/JDK-7025832
+        assertThat(higher).isLessThan(lower);
```

Comment:
What's the purpose of testing the JDK implementation of `isLessThan`? I don't see anything below that relies on this behavior.

Reply:
But we don't use Java UUIDs for ordering, so how is that behavior relevant? If anything, such documentation should be added to the UUID type class so that anyone using it in their connector is aware. This comment is going to be largely invisible for that purpose.

Reply:
Given that we have `javaUuidToTrinoUuid`/`trinoUuidToJavaUuid` public methods in `UuidType`, that information seems relevant for anyone manipulating UUIDs, no?

Reply:
I agree that there can be some other place where the information would be useful _too_.
it doesn't make it unnecessary in the _uuid ordering test_.

Reply:
As somebody who tends to look at unit tests quite often when getting an understanding of things, I do appreciate this form of documentation for something a bit... quirky / not intuitively obvious when looking at the UUID type class directly.

This could possibly be documented elsewhere as mentioned, but I agree with @findepi that it has utility as a unit test. Particularly, the rest of the test seems to testing that two Trino UUID's are comparable as well as how to do it. If I were working on a feature (say for the Iceberg connector), this is amongst the first places I would likely go for looking at using the code.

If this isn't going to be user facing documentation, then I think this comparison test for the JDK UUIDs is beneficial. Though a comment in `UuidType` would also suffice for me - but that's more up to you guys. I think the rest of the test is beneficial though.